In [2]:
import pandas as pd
import matplotlib.pyplot as plt

# Load data and overview

In [18]:
data = pd.read_csv('data/heart.csv')


In [9]:
data.sample(10)

(303, 14)


age  sex  cp  trtbps  chol  fbs  restecg  thalachh  exng  oldpeak  slp  \
138   57    1   0     110   201    0        1       126     1      1.5    1   
125   34    0   1     118   210    0        1       192     0      0.7    2   
109   50    0   0     110   254    0        0       159     0      0.0    2   
214   56    1   0     125   249    1        0       144     1      1.2    1   
301   57    1   0     130   131    0        1       115     1      1.2    1   
140   51    0   2     120   295    0        0       157     0      0.6    2   
286   59    1   3     134   204    0        1       162     0      0.8    2   
263   63    0   0     108   269    0        1       169     1      1.8    1   
98    43    1   2     130   315    0        1       162     0      1.9    2   
202   58    1   0     150   270    0        0       111     1      0.8    2   

     caa  thall  output  
138    0      1       1  
125    0      2       1  
109    0      2       1  
214    1      2       0  
301    1      3       0  
140    0      2       1  
286    2      2       0  
263    2      2       0  
98     1      2       1  
202    0      3       0

In [10]:
# data types
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 303 entries, 0 to 302
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       303 non-null    int64  
 1   sex       303 non-null    int64  
 2   cp        303 non-null    int64  
 3   trtbps    303 non-null    int64  
 4   chol      303 non-null    int64  
 5   fbs       303 non-null    int64  
 6   restecg   303 non-null    int64  
 7   thalachh  303 non-null    int64  
 8   exng      303 non-null    int64  
 9   oldpeak   303 non-null    float64
 10  slp       303 non-null    int64  
 11  caa       303 non-null    int64  
 12  thall     303 non-null    int64  
 13  output    303 non-null    int64  
dtypes: float64(1), int64(13)
memory usage: 33.3 KB


We have 303 samples as well as 14 variables.  
There's no missing values.  
The target variable is `output`

In [21]:
data.describe()

age         sex          cp      trtbps        chol         fbs  \
count  302.00000  302.000000  302.000000  302.000000  302.000000  302.000000   
mean    54.42053    0.682119    0.963576  131.602649  246.500000    0.149007   
std      9.04797    0.466426    1.032044   17.563394   51.753489    0.356686   
min     29.00000    0.000000    0.000000   94.000000  126.000000    0.000000   
25%     48.00000    0.000000    0.000000  120.000000  211.000000    0.000000   
50%     55.50000    1.000000    1.000000  130.000000  240.500000    0.000000   
75%     61.00000    1.000000    2.000000  140.000000  274.750000    0.000000   
max     77.00000    1.000000    3.000000  200.000000  564.000000    1.000000   

          restecg    thalachh        exng     oldpeak         slp         caa  \
count  302.000000  302.000000  302.000000  302.000000  302.000000  302.000000   
mean     0.526490  149.569536    0.327815    1.043046    1.397351    0.718543   
std      0.526027   22.903527    0.470196    1.161452    0.616274    1.006748   
min      0.000000   71.000000    0.000000    0.000000    0.000000    0.000000   
25%      0.000000  133.250000    0.000000    0.000000    1.000000    0.000000   
50%      1.000000  152.500000    0.000000    0.800000    1.000000    0.000000   
75%      1.000000  166.000000    1.000000    1.600000    2.000000    1.000000   
max      2.000000  202.000000    1.000000    6.200000    2.000000    4.000000   

            thall      output  
count  302.000000  302.000000  
mean     2.314570    0.543046  
std      0.613026    0.498970  
min      0.000000    0.000000  
25%      2.000000    0.000000  
50%      2.000000    1.000000  
75%      3.000000    1.000000  
max      3.000000    1.000000

## Any duplicates

In [19]:
# we have one duplicated sample
print(data.duplicated().sum())


1


In [20]:
# remove duplicate
data = data.drop_duplicates()
data.shape

(302, 14)

## Correlation